contrib/OtlpGrpc/SpanConverter.php

```diff
@@ -81,25 +81,23 @@ public function as_otlp_span_kind($kind): int
         return SpanKind::SPAN_KIND_UNSPECIFIED;
     }
 
-    public function as_otlp_span(ReadableSpan $span): CollectorSpan
+    public function as_otlp_span(SpanData $span): CollectorSpan
     {
-        $end_timestamp = ($span->getStartEpochTimestamp() + $span->getDuration());
-
-        $parent_span = $span->getParent();
-        $parent_span_id = $parent_span ? $parent_span->getSpanId() : false;
+        $parent_span = $span->getParentContext();
+        $parent_span_id = $parent_span->isValid() ? $parent_span->getSpanId() : null;
 
         $row = [
-            'trace_id' => hex2bin($span->getContext()->getTraceId()),
-            'span_id' => hex2bin($span->getContext()->getSpanId()),
+            'trace_id' => hex2bin($span->getTraceId()),
+            'span_id' => hex2bin($span->getSpanId()),
             'parent_span_id' => $parent_span_id ? hex2bin($parent_span_id) : null,
-            'name' => $span->getSpanName(),
-            'start_time_unix_nano' => $span->getStartEpochTimestamp(),
-            'end_time_unix_nano' => $end_timestamp,
-            'kind' => $this->as_otlp_span_kind($span->getSpanKind()),
+            'name' => $span->getName(),
+            'start_time_unix_nano' => $span->getStartEpochNanos(),
+            'end_time_unix_nano' => $span->getEndEpochNanos(),
+            'kind' => $this->as_otlp_span_kind($span->getKind()),
             'trace_state' => (string) $span->getContext()->getTraceState(),
-            'dropped_attributes_count' => $span->getAttributes()->getDroppedAttributesCount(),
-            'dropped_events_count' => $span->getDroppedEventsCount(),
-            'dropped_links_count' => $span->getDroppedLinksCount(),
+            'dropped_attributes_count' => $span->getTotalDroppedAttributes(),
+            'dropped_events_count' => $span->getTotalDroppedEvents(),
+            'dropped_links_count' => $span->getTotalDroppedLinks(),
```

Comment:
Technically these should not be set if they are `0`. But this can be handled in another PR.